src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,289 @@
+# pylint: disable=wrong-import-order
+
+from __future__ import annotations
+
+from typing import Any
+import numpy as np
+import pandas as pd
+import regex
+from unittest import mock
+
+from dsp_tools.models.exceptions import UserError
+
+languages = ["en", "de", "fr", "it", "rm"]
+
+
+def read_excel_file(excel_filename: str) -> pd.DataFrame:
+    """
+    This function reads an Excel file, if there is a ValueError then it patches the openpyxl part that creates the
+    error and opens it with that patch. It cleans and then returns the pd.DataFrame.
+
+    Args:
+        excel_filename: The name of the Excel file
+
+    Returns:
+        A pd.DataFrame
+    """
+    # load file
+    try:
+        read_df: pd.DataFrame = pd.read_excel(excel_filename)
+    except ValueError:
+        # Pandas relies on openpyxl to parse XLSX files.
+        # A strange behaviour of openpyxl prevents pandas from opening files with some formatting properties
+        # (unclear which formatting properties exactly).
+        # Apparently, the excel2json test files have one of the unsupported formatting properties.
+        # Credits: https://stackoverflow.com/a/70537454/14414188
+        with mock.patch("openpyxl.styles.fonts.Font.family.max", new=100):
+            read_df = pd.read_excel(excel_filename)
+    read_df = clean_data_frame(unclean_df=read_df)
+    return read_df
+
+
+def clean_data_frame(unclean_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This function takes a pd.DataFrame and removes:
+        - Leading and trailing spaces from the column names
+        - Leading and trailing spaces from each cell
+        - Any characters in the cells that are not part of any known language, for example, linebreaks and replaces it
+          with a pd.NA.
+        - Removes all rows that are empty in all columns
+
+    Args:
+        unclean_df: The pd.DataFrame that is to be cleaned
+
+    Returns:
+        pd.DataFrame which has the above-mentioned removed
+    """
+    # Remove leading and trailing blanks in column names and make them lower case
+    cleaned_df = unclean_df.rename(columns=lambda x: x.strip().lower())
+    # Remove the values of all cells that do not at least contain one character of any known language
+    cleaned_df = cleaned_df.applymap(
+        lambda x: str(x).strip() if pd.notna(x) and regex.search(r"[\w\p{L}]", str(x), flags=regex.U) else pd.NA
+    )
+    # drop all the rows that are entirely empty
+    cleaned_df.dropna(axis=0, how="all", inplace=True)
+    return cleaned_df
+
+
+def check_required_columns_raise_error(check_df: pd.DataFrame, required_columns: set[str]) -> None:
```

Comment:
If I try to describe what this function does, I would say it checks a dataframe if it contains the required columns. So I would name it `check_that_df_contains_required_columns()`